opportunity/src/app/data/services/firebase.service.ts

```diff
@@ -26,14 +30,69 @@ export class FirebaseCrudService {
       }))
     );
   }
+  
+  registerUser(userLoginData: { logInEmail: string, photoURL: string, displayName: string }, isNgo: boolean) {
+    console.log(isNgo);
+    const { logInEmail, photoURL, displayName } = userLoginData;
+    let subject$ = new Subject();
+    const vol$ = this.getMany('volunteers', (ref) => ref.where('username', '==', logInEmail));
+    const ngo$ = this.getMany('ngos', (ref) => ref.where('username', '==', logInEmail));
+    combineLatest(
+      vol$,
+      ngo$,
+      (vol, ngo) => vol.length ? { type: 'volunteer', data: vol[0] } : ngo.length ? { type: 'ngo', data: ngo[0] } : { type: '404' }
+    ).subscribe(async (res) => {
+      console.log(res);
+      switch(res.type) {
+        case 'volunteer':
+          subject$.next({
+            isNgo: false,
+            user: res.data,
+          })
+          break;
+        case 'ngo':
+          subject$.next({
+            isNgo: true,
+            user: res.data,
+          })
+          break;
+        case '404':
```

Comment:
please separate the reg and auth stuff into a separate service as the file gets amazingly long